### PR TITLE
add bids download to master

### PR DESCRIPTION
**Pull request by rentzso**
This is a port of bids downloads to the master branch from aloha.
Minimal adaptation was done for standard downloads, but they essentially work as before.

### CODE REVIEW

**Comment by rentzso** [comment]
closes #78

File: api/core.py [hunk 1]
```diff
@@ -245,12 +299,17 @@ def download(self):
...
             self.response.headers['Content-Disposition'] = 'attachment; filename=' + str(ticket['filename'])
+            for project_id in ticket['projects']:
+                self.app.db.projects.update_one({'_id': project_id}, {'$inc': {'counter': 1}})
         else:
             req_spec = self.request.json_body
             validator = validators.payload_from_schema_file(self, 'input/download.json')
             validator(req_spec, 'POST')
             log.debug(json.dumps(req_spec, sort_keys=True, indent=4, separators=(',', ': ')))
-            return self._preflight_archivestream(req_spec)
+            if self.request.GET.get('format') == 'bids':
```
Comment (gsfr) [inline api/core.py]: Use `self.get_param()`?

**Comment by rentzso** [comment]
will work on this pull request to add a generic way of filtering downloads by type or tag.

**Comment by rentzso** [comment]
Still need to address Gunnar's comment, apart from that the PR is ready to be reviewed.

**Comment by rentzso** [comment]
addressed @gsfr comment

File: api/core.py [hunk 2]
```diff
@@ -19,6 +19,64 @@
...
+            #
+            # download_payload = {
+            #     'optional': True,
+            #     'nodes': [{'level':'project', '_id':project_id}],
+            #     'filters':[{
+            #         'tags':{'+':['incomplete']},
+            #         'types':{'+':['dicom']}
+            #     }]
+            # }
+            # will download only files with tag 'incomplete' AND type different from 'dicom'
```
Comment (gsfr) [inline api/core.py]: Wouldn't this long comment be better placed into the docstring of the download method?

**Comment by gsfr** [comment]
LGTM.

The branch is quite a bit behind master, so please rebase before the merge to make sure we don't get any unintended side-effects.

**Pull request merged**